src/titlepane/TitlePane.ts

```diff
@@ -0,0 +1,127 @@
+import uuid from '@dojo/core/uuid';
+import { DNode } from '@dojo/widget-core/interfaces';
+import { theme, ThemeableMixin, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import { v } from '@dojo/widget-core/d';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+
+import { Keys } from '../common/util';
+
+import * as css from './styles/titlePane.m.css';
+
+/**
+ * @type TitlePaneProperties
+ *
+ * Properties that can be set on a TitlePane component
+ *
+ * @property closeable          If false the pane will not collapse in response to clicking the title
+ * @property headingLevel       'aria-level' for the title's DOM node
+ * @property onRequestClose     Called when the title of an open pane is clicked
+ * @property onRequestOpen      Called when the title of a closed pane is clicked
+ * @property open               If true the pane is opened and content is visible
+ * @property title              Title to display above the content
+ */
+export interface TitlePaneProperties extends ThemeableProperties {
+	closeable?: boolean;
+	headingLevel?: number;
+	onRequestClose?(titlePane: TitlePane): void;
+	onRequestOpen?(titlePane: TitlePane): void;
+	open?: boolean;
+	title: string;
+};
+
+export const TitlePaneBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class TitlePane extends TitlePaneBase<TitlePaneProperties> {
+	private _contentId = uuid();
+	private _titleId = uuid();
+
+	private _afterRender(element: HTMLElement) {
+		// Conditionally adjust top margin. Done manually instead of through Maquette
+		// so the underlying DOM is accessible, as we need to know the content height.
+		// Put in a rAF to push this operation to the next tick, otherwise
+		// element.offsetHeight can be incorrect (e.g. before styling is applied)
+		// Note that this will go away when meta support is added to widget-core
+		requestAnimationFrame(() => {
+			const { open = true } = this.properties;
```

Comment:
No, I'm not using `max-height` for the animation. We're using `margin` so we could avoid the bugs that plagued using `max-height`, it was a trick I discovered while working on this a while ago. The animations you mention were leftover and have been removed from this PR.